parm/metplus/separate_fcst_obs_info.py

```diff
@@ -0,0 +1,249 @@
+                    # observations.
+                    threshes_fcst.append(thresh_fcst)
+                    threshes_obs.append(thresh_obs)
+    
+                print(f'            threshes_fcst = {threshes_fcst}')
+                print(f'            threshes_obs = {threshes_obs}')
+    
+                # Set the key-value pair consisting of the current forecast level (the key)
+                # and corresponding list of forecast thresholds (the value) in the levels-
+                # and-thresholds forecast dictionary of the current field.  Then do the
+                # same for the observations.
+                levels_threshes_fcst[level_fcst] = threshes_fcst
+                levels_threshes_obs[level_obs] = threshes_obs
+    
+            print(f'        levels_threshes_fcst = {levels_threshes_fcst}')
+            print(f'        levels_threshes_obs = {levels_threshes_obs}')
+    
+            # Set the key-value pair consisting of the current forecast field name (the
+            # key) and corresponding forecast levels-and-thresholds forecast dictionary
+            # (the value) in the fields-and-levels-and-thresholds forecast dictionary.
+            # Then do the same for the observations.
+            #fields_levels_threshes_fcst[field_fcst] = levels_threshes_fcst
+            #fields_levels_threshes_obs[field_obs] = levels_threshes_obs
+            fields_levels_threshes_fcst.append({field_fcst: levels_threshes_fcst})
+            fields_levels_threshes_obs.append({field_obs: levels_threshes_obs})
+    
+        print(f'    fields_levels_threshes_fcst = {fields_levels_threshes_fcst}')
+        print(f'    fields_levels_threshes_obs = {fields_levels_threshes_obs}')
+
+        # Set the key-value pair consisting of the current ...
+        # Then do the same for the observations.
+        #fgs_fields_levels_threshes_fcst.append(fields_levels_threshes_fcst)
+        #fgs_fields_levels_threshes_obs.append(fields_levels_threshes_obs)
+        fgs_fields_levels_threshes_fcst[field_group] = fields_levels_threshes_fcst
+        fgs_fields_levels_threshes_obs[field_group] = fields_levels_threshes_obs
+
+    print(f'')
+    print(f"""\nfgs_fields_levels_threshes_fcst =\n{get_pprint_str(fgs_fields_levels_threshes_fcst, '  ')}\n""")
+    print(f'')
+    print(f"""\nfgs_fields_levels_threshes_obs =\n{get_pprint_str(fgs_fields_levels_threshes_obs, '  ')}\n""")
+
+    # Combine forecast and obs verfication dictionaries into one and
+    # write to a yaml file.
+    vx_config_dict = {'fcst': fgs_fields_levels_threshes_fcst,
+                      'obs': fgs_fields_levels_threshes_obs}
+    dict_to_str = get_pprint_str(vx_config_dict, '  ')
+    # Convert the dictionary of jinja variable settings above to yaml format
+    # and write it to a temporary yaml file for reading by the set_template
+    # function.
+    filename = ''.join(['tmp.vx_config_', det_or_ens, '_dict.split_fcst_obs.txt'])
```

Comment:
I have a few concerns with how this file is being handled. This doesn't seem to actually be used as a "temporary" file, since it is read by potentially many different verification tasks. And given that it is used at run time, it should be placed in the experiment directory, not kept in this directory. Additionally, if it is a yaml file meant to be read in as such, it should have the suffix `.yaml`